### PR TITLE
[CMDCT-4555]: Update dateOptional schema to handle untrue dates

### PR DESCRIPTION
**Pull request by ntsummers1**
### Description

Optional date fields were allowing strings like "42/42/4242" in them, which isn't a real date. This fixes that!

https://github.com/user-attachments/assets/62ddaf93-4f42-4b6e-9a32-19fe68486a65

### Related ticket(s)

CMDCT-4555

---
### How to test

You'll need to fillout the form all the way to plan compliance so you can view the dateoptional field. Inside there you'll be able to play around with it.

---
### Pre-review checklist

- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [x] I have updated relevant documentation, if necessary
- [x] I have performed a self-review of my code
- [x] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

### CODE REVIEW

File: services/ui-src/src/utils/validation/schemas.ts [hunk 1]
```diff
@@ -235,3 +226,5 @@ export const nested = (
 // REGEX
 export const dateFormatRegex =
   /^((0[1-9]|1[0-2])\/(0[1-9]|1\d|2\d|3[01])\/(19|20)\d{2})|((0[1-9]|1[0-2])(0[1-9]|1\d|2\d|3[01])(19|20)\d{2})$/;
+export const optionalDateFormatRegex =
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/utils/validation/schemas.ts:229]: We could use the same regex for both like this:

```
const datePattern =
  "((0[1-9]|1[0-2])\\/(0[1-9]|1\\d|2\\d|3[01])\\/(19|20)\\d{2})|((0[1-9]|1[0-2])(0[1-9]|1\\d|2\\d|3[01])(19|20)\\d{2})";

export const dateFormatRegex = new RegExp(`^${datePattern}$`);
export const optionalDateFormatRegex = new RegExp(`^(${datePattern})?$`);

```

File: services/ui-src/src/utils/validation/schemas.test.ts [hunk 2]
```diff
@@ -69,20 +69,15 @@ describe("Schemas", () => {
...
   ];
 
-  const goodDateOptionalTestCases = [
-    "",
-    null,
-    undefined,
-    "01/01/2023",
-    "05/15/2023",
-  ];
+  const goodDateOptionalTestCases = ["", "01/01/2023", "05/15/2023"];
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/utils/validation/schemas.test.ts:72]: Why remove the `null` and `undefined` test cases?

Comment (ntsummers1) [reply services/ui-src/src/utils/validation/schemas.test.ts:72]: `.nullable()` was previously being used in all of our validations when working with optional fields. We later learned that this was unnecessary to do, and was causing some weird edge cases. Dateoptional never received the update, and thus I've removed it from the validation.

**Review by bangbay-bluetiger — APPROVED** [review]

**Review by gmrabian — APPROVED** [review]
